File: csharp/demo/demo02b-join.cs
/*
 * THREAD JOINS
 */
using System;
using System.Threading;



class Demo02B : IRunnable
{
    public void run()
    {
        Thread thFoo = new Thread(() => Console.WriteLine("foo"));
        Thread thBar = new Thread(() => Console.WriteLine("bar"));

        thFoo.Start();
        thBar.Start();

        // thFoo.Join();
        // thBar.Join();

        /*
         * We do not need to call thFoo.Join() and thBar.Join().
         * The reason is main thread will wait for the completion of all threads before app exits.
         */
    }
}

File: python/demo02b_join.py
'''
THREAD JOINS
'''

import threading



th_foo = threading.Thread(target=lambda:print('foo'))
th_bar = threading.Thread(target=lambda:print('bar'))

th_foo.start()
th_bar.start()

# th_foo.join()
# th_bar.join()

'''
We do not need to call th_foo.join() and th_bar.join().
The reason is main thread will wait for the completion of all threads before app exits.
'''
